lib/cartopy/mpl/geoaxes.py

```diff
@@ -427,10 +427,45 @@ def draw(self, renderer=None, inframe=False):
                 self.imshow(img, extent=extent, origin=origin,
                             transform=factory.crs, *args[1:], **kwargs)
         self._done_img_factory = True
-
+        self._cachedRenderer = renderer
         return matplotlib.axes.Axes.draw(self, renderer=renderer,
                                          inframe=inframe)
 
+    def _update_title_position(self, renderer):
+        """
+        Update the title position based on the bounding box enclosing
+        all the ticklabels and x-axis spine and xlabel...
+
+        """
+        matplotlib.axes.Axes._update_title_position(self, renderer)
+        if not self._gridliners:
+            return
+
+        if self._autotitlepos is not None and not self._autotitlepos:
```

Comment:
I don't quite understand what this statement checks for; can you explain please?

Reply:
I also comes from matplotlib:
https://github.com/matplotlib/matplotlib/blob/529410fa0c9379cd60b1e4d8fdd0478bb2f7e892/lib/matplotlib/axes/_base.py#L2471

I just remove the logging call.

Reply:
OK, fair enough.  I still don't understand, but I trust you.